lib/rules/named-grid-areas-no-invalid/utils/isRectangular.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+/**
+ *
+ * @param {string[][]} areas
+ * @returns {boolean}
+ */
+function isRectangular(areas) {
+	const columnsPerRow = areas.map((row) => row.length);
+
+	return columnsPerRow.every((val, i, arr) => val === arr[0]);
```

Comment:
Does the following simplification make sense?

```suggestion
	return areas.every((row, i, arr) => row.length === arr[0].length);
```